tests/src/test/scala/org/apache/openwhisk/core/database/test/BatcherTests.scala

```diff
@@ -123,4 +124,71 @@ class BatcherTests extends FlatSpec with Matchers with WskActorSystem {
     an[Exception] should be thrownBy await(r3)
     an[Exception] should be thrownBy await(r4)
   }
+
+  it should "complete batched values with max retry limit" in {
+    val p = Promise[Unit]()
+
+    val maxRetry = 3
+    val batchSize = 1
+    val concurrency = 1
+
+    var retryCount = new AtomicInteger(0)
+
+    def doStore(els: Seq[Int], retry: Int): Future[Seq[Int]] = {
+      val result = if (retry > 0) {
+        Future.failed(new Exception)
+      } else {
+        p.future.map(_ => els)
+      }
+
+      result.recoverWith {
+        case _ if retry > 0 =>
+          retryCount.incrementAndGet()
+          doStore(els, retry - 1)
+        case e =>
+          Future.failed(e)
+      }
+
+    }
+    val batcher = new Batcher[Int, Int](batchSize, concurrency, maxRetry)(doStore)
+
+    val values = List(1)
+    val results = values.map(batcher.put)
+
+    p.success(())
+
+    await(Future.sequence(results)) shouldBe values
+
+    retryCount.get() shouldBe maxRetry
+  }
+
+  it should "complete batched values with the thrown exception with max retry limit" in {
```

Comment:
Actually, have no method to test `ElasticSearchActivationStore.scala's doStore` method in unit test case.
I just wrote a similar retry logic method here, and tested it.

If want to make  `ElasticSearchActivationStore.scala's doStore`  can be tested, need to change a lot.